src/view/filter.rs

```diff
@@ -5,23 +5,31 @@ use crate::{Direction, LinkView, MultiView, NodeIndex, PortIndex, PortView};
 use context_iterators::{ContextIterator, FilterWithCtx, IntoContextIterator};
 use delegate::delegate;
 
-/// Node filter used by [`NodeFiltered`].
+/// Node filter used by [`NodeFiltered`] and [`PortFiltered`].
 pub type NodeFilter<Ctx> = fn(NodeIndex, &Ctx) -> bool;
+/// Port filter used by [`PortFiltered`].
+pub type PortFilter<Ctx> = fn(PortIndex, &Ctx) -> bool;
 
 /// A wrapper around a portgraph that filters out nodes.
 #[derive(Debug, Clone, PartialEq)]
-pub struct NodeFiltered<'a, G, F, Context = ()> {
+pub struct PortFiltered<'a, G, FN, FP, Context = ()> {
```

Comment:
I think this should rather be called `LinkFiltered` (or `NodeLinkFiltered` (or just `Filtered`?))
since it's not actually filtering the ports.

At least the doc should document the behaviour.

Reply:
Sure, how about `FilteredGraph`?